### ISSUE TITLE
Microstrip k_c: evaluate the incremental-inductance derivative instead of Wheeler's 1942 fit

### ISSUE THREAD

**Issue by gvcallen** [paramrf/paramrf#120]
## What to build

A second microstrip `AbstractCurrentDistribution` that evaluates Wheeler's incremental-inductance **rule** directly, by autodiff, instead of his 1942 closed-form fit to it. `WheelerCurrentDistribution` stays available — it is the cited standard, and it is what reproduces published results and matches ADS/scikit-rf.

### The defect

`WheelerCurrentDistribution._distribute` (`pmrf/models/components/lines/microstrip.py:83`) computes the conductor geometry weight as

```python
weight = 2 / cross_section.w * jnp.exp(-1.2 * (jnp.real(zc) / z0) ** 0.7)
```

The `2/W` prefactor is right. The empirical `exp[-1.2 (Re Zc / Z_0)^0.7]` edge-crowding factor is not: it over-predicts `k_c` by **22% at `w/h = 2.5`**, falling to 4% at `w/h = 50`. Because the error varies with geometry by a factor of four across that range, no scale factor patches it — it has to be computed.

This is worse than it looks. The conductor channel enters `Z` only through the product `Z_s * k_c`, and `Z_s ~ sigma^(-1/2)`, so `k_c` and conductivity are exactly degenerate. A fit with free `sigma` absorbs the whole error silently: a 22% `k_c` error surfaces as `1.22^2 = 1.49x` copper. The user never sees a bad `k_c`, only an implausible conductivity.

### The fix

Evaluate the rule:

$$k_c = \frac{1}{\eta_0}\frac{\partial}{\partial n}Z_a(W-2n,\ H+2n,\ T-2n)\bigg|_{n=0}$$

`Z_a` is the same cross-section with the dielectric removed. Every conductor surface recedes into the metal by `n`: the strip narrows by `2n` and thins by `2n`, and the strip-to-ground gap grows by `2n` (`n` from the ground plane receding, `n` from the strip's underside).

The distribution needs a formulation, which `_distribute(freq, cross_section, quasi_static)` does not receive. **Give the new distribution its own `formulation` field** rather than widening the `_distribute` signature — smaller change, and it makes explicit which `Z_a` the derivative is taken on, which matters (see the trap).

Sketch, verified to run against the current tree:

```python
class IncrementalInductanceCurrentDistribution(AbstractCurrentDistribution[MicrostripCrossSection]):
    cross_section_type: ClassVar[type] = MicrostripCrossSection

    formulation: AbstractMicrostripFormulation = eqx.field(
        default_factory=HammerstadJensenMicrostripFormulation
    )
    slab_impedance: AbstractSurfaceImpedance = eqx.field(
        default_factory=RootSumSquareSlabSurfaceImpedance
    )

    def _distribute(self, freq, cross_section, quasi_static):
        z0 = jnp.sqrt(mu_0 / epsilon_0)
        w, h, t = cross_section.w, cross_section.h, cross_section.t

        def z_air(n):
            result = self.formulation.quasi_static(
                w=w - 2 * n, h=h + 2 * n,
                t=None if t is None else t - 2 * n,
                ep_r=jnp.ones_like(quasi_static.ep_eff),
            )
            return jnp.real(result.zc)[0]

        weight = jax.grad(z_air)(0.0) / z0
        impedance = HalfSpaceSurfaceImpedance() if t is None else self.slab_impedance
        return ((impedance, weight),)
```

Notes:

- **Use `jax.grad`, not a finite difference.** `Z_a` is closed form, so autodiff is exact and free. A finite difference also has no good step size: the derivative is ~1.4e5 ohm/m against a `Z_a` of ~77 ohm, so a small step loses precision and a large one picks up curvature. (Autodiff and a central difference at `n = 1e-6` were checked to agree to five significant figures at every `w/h` below.)
- `ep_r = 1` gives `eps_eff = 1`, so `quasi_static(...).zc` *is* `Z_a`. Confirmed against an independent field solve.
- The weight is real and frequency-independent, exactly as Wheeler's is, and stays differentiable in `w`, `h` and `t` — which the fitting path needs.
- `jnp.real(...)[0]` assumes `zc` carries a frequency axis; follow the surrounding code rather than the sketch literally.

## The trap: the formulation must be thickness-aware

The derivative is taken on the supplied formulation's own `Z_a`. A formulation that ignores `t` cannot respond to the strip thinning, so the top face and the sidewalls are simply absent from the weight.

At `w/h = 2.5` (`w` = 4 mm, `h` = 1.6 mm, `t` = 35 um):

| pairing | `k_c` [1/m] |
|---|---:|
| Hammerstad–Jensen, `t = 35 um` | **318.34** |
| Hammerstad–Jensen, `t = None` | 219.38 |
| `WheelerMicrostripFormulation` (1977), `t = 35 um` | 221.39 |
| `WheelerCurrentDistribution` (the 1942 fit being replaced) | 385.69 |

`WheelerMicrostripFormulation.quasi_static` accepts and ignores `t` by construction (`microstrip.py:148`), so it lands in exactly this state. An independent field solve that meshes the real 35 um bar gives 316. Pairing the new distribution with a thickness-blind formulation, or leaving `t` unspecified, under-predicts by 30% — an error of similar size to the one being fixed, in the opposite direction, and just as invisible behind a free `sigma`.

**Either raise on a thickness-blind formulation or document it prominently. Do not let it default into that state silently.**

## Acceptance test

Reproducible with ParamRF alone. Constant `ep_r = 4.335`, `h` = 1.6 mm, `t` = 35 um, Hammerstad–Jensen, x64 enabled. `Re(Zc)` is of the dielectric-filled line, which is what Wheeler's fit consumes. Every row below was re-run against the current tree and reproduces to the digits shown.

| `w/h` | `w` [mm] | `Re(Zc)` | Wheeler 1942 | corrected | ratio |
|---:|---:|---:|---:|---:|---:|
| 2.5 | 4.0 | 42.28 | 385.69 | 318.34 | 0.8254 |
| 5.0 | 8.0 | 25.96 | 207.89 | 178.62 | 0.8592 |
| 10.0 | 16.0 | 14.83 | 110.35 | 98.47 | 0.8923 |
| 20.0 | 32.0 | 8.06 | 57.62 | 53.24 | 0.9241 |
| 50.0 | 80.0 | 3.43 | 23.91 | 22.91 | 0.9581 |

### Do not use `k_c -> 2/W` as an acceptance test

The wide-line limit does **not** recover Wheeler's `2/W` prefactor. At `w/h = 50`, `k_c` = 22.91 against `2/W` = 25.0, a ratio of 0.916. The `2/W` argument assumes a zero-thickness strip and neglects fringing; with fringing `C_air` is larger and `k_c = -eps_0/C^2 * dC/dn` falls below it. Both `Z_a` routes agree there, so this is physics, not a solver artifact.

## External validation

These cannot be reproduced inside ParamRF; recorded so the corrected column can be trusted.

**Against an independent 2D quasi-static field solver.** The same recession derivative taken numerically on an air-filled electrostatic solve that meshes the actual cross-section and knows nothing about Hammerstad–Jensen. Converged in both mesh and step size:

| `w/h` | `Z_a` field | `Z_a` HJ | diff | `k_c` field | `k_c` HJ | HJ / field |
|---:|---:|---:|---:|---:|---:|---:|
| 2.5 | 76.593 | 77.326 | +0.96% | 316.04 | 318.34 | 1.0073 |
| 5.0 | 48.841 | 49.063 | +0.46% | 177.33 | 178.62 | 1.0073 |
| 10.0 | 28.854 | 28.919 | +0.23% | 97.72 | 98.47 | 1.0076 |
| 20.0 | 16.076 | 16.093 | +0.11% | 52.95 | 53.24 | 1.0056 |
| 50.0 | 6.980 | 6.984 | +0.05% | 22.85 | 22.91 | 1.0025 |

Two independent `Z_a` routes agree on `k_c` to within 0.76%. Two further routes agree at `w/h = 2.5`: a power-loss surface-impedance integral `k_c = (Int rho_s^2 dl)/Q^2` on the same air-filled problem converges to 310–320, and a volumetric PEEC solve with no surface-impedance approximation anywhere gives an equivalent `k_c` of ~306. All four sit in 306–320 against Wheeler's 385.7.

**Against a commercial 3D solver.** At that solver's exact inputs (`w` = 4 mm, `h` = 1.6 mm, `t` = 35 um, `ep_r` = 4.335, `tand` = 0.0183, `sigma` = 5.8e7), zero free parameters, against its exported port-mode propagation constant over 50–130 MHz:

| `k_c` source | mean alpha err | band-RMS alpha err | mean beta err |
|---|---:|---:|---:|
| Wheeler 1942 (shipped) | +9.97% | 10.04% | +0.018% |
| incremental inductance | +4.49% | 4.53% | −0.040% |

Band-RMS attenuation error more than halves, at a cost of 0.06 percentage points on `beta`. `Z_c` is unaffected — the conductor term is worth 0.65% of `Z`. A residual remains against that solver; it has been traced to the external tool and is not a ParamRF matter. This change is not expected to close it.

## Relationship to #101, and an open disagreement

#101 corrects the *same* Wheeler geometry weight by a different route: splitting it into separate trace and ground-plane terms. The two are **independent peers** — neither blocks the other, and **neither changes the microstrip default**, which stays `WheelerCurrentDistribution` until the following is settled.

They disagree on the sign of Wheeler's error:

| source | Wheeler's error vs. the better answer |
|---|---|
| this ticket | Wheeler is **22% high** at `w/h = 2.5`, corroborated by four independent routes landing in 306-320 against Wheeler's 385.7 |
| #101 | Wheeler is **12-18% low** at strong skin, consistent with Holloway and Kuester's reported 12-30% underprediction |

Both cite Holloway and Kuester; they cannot both be right. This ticket does not have to resolve it, but it must not pretend the disagreement is absent: whoever implements this should state the departure from #101's split **with its sign** in the docstring rather than describing the derivative as simply "more accurate". #103 carries the standing record of the contradiction.

## Acceptance criteria

- [ ] `IncrementalInductanceCurrentDistribution` exists as a peer of `WheelerCurrentDistribution`, carrying its own `formulation` field; `WheelerCurrentDistribution` remains available and unchanged
- [ ] The weight is computed with `jax.grad`, not a finite difference
- [ ] Pairing with a thickness-blind formulation, or an unspecified `t`, either raises or is documented prominently — not silent
- [ ] The five-row acceptance table above is reproduced as a test, with per-case tolerances
- [ ] A test asserts the wide-line limit does **not** approach `2/W`, with the reason recorded
- [ ] The weight stays differentiable in `w`, `h` and `t`
- [ ] Docstring carries a `**Mathematical Formulation**` section stating the recession derivative and a `References` section citing Wheeler (1942), per repo convention, and distinguishes the rule from the 1942 fit and from `WheelerMicrostripFormulation` (1977)

## Notes for whoever picks this up

- Written against 0.34.1; every ParamRF number in it was re-verified against 0.34.3 at `c1386448` and reproduces exactly.
- The original handoff's sketch used `slab_shape`, `RootSumSquareSlabShape` and `HalfSpaceShape`. Those are now `slab_impedance`, `RootSumSquareSlabSurfaceImpedance` and `HalfSpaceSurfaceImpedance` (renamed in 52b75007); the sketch above is corrected.
- Related, both **reopened** on 2026-09-04 after it emerged they were closed in a four-issue bulk sweep (#91, #101, #102, #103, one second apart, no comments) without landing: #101 (trace/ground split, now an independent peer of this ticket) and #103 (documenting the conductor-loss alternatives, rescoped to carry the sign contradiction). #121 holds the surface-impedance documentation point split out of this ticket.
- #91 and #102 were swept in the same bulk close and also did not land — stripline still has no dc resistance floor and returns a zero weight at `t=None` (`stripline.py:72-86`). Not addressed here; flagged so it is not lost.
